File: examples/disk-IO-examples/GenerateText.java
import java.io.BufferedWriter;
import java.io.FileWriter;
import java.io.IOException;
import java.io.PrintWriter;
import java.util.Random;

public class GenerateText
{
	final static int MAX_KEY = 1000000;
	final static int MAX_DOUBLE = 2148;

    private static void generateFile(int n, long seed, PrintWriter out)
    {
		int i;
		int key;
		double value1;
		double value2;
		double value3;
		Random generator = new Random(seed);
	
		for (i=0; i<n; i++) {
			/*key = generator.nextInt(MAX_KEY); */
			key = 2 * i;
			value1 = generator.nextDouble() * MAX_DOUBLE;
			value2 = generator.nextDouble() * MAX_DOUBLE;
			value3 = generator.nextDouble() * MAX_DOUBLE;
			out.println(key+" " + value1 + " " + value2 + " " + value3);
		}
	}

	public static void main(String argv[])
	{
		int n;
		long seed = 0;

		if (argv.length < 1) {
			System.err.println("Usage: java GenerateText <n> [<seed>]\n");
			System.exit(1);
		}

		n = Integer.parseInt(argv[0]);
	    if (argv.length == 2) {
			seed = Long.parseLong(argv[1]);
		} 


		try {
			FileWriter fout = new FileWriter("data.txt");
			PrintWriter out = new PrintWriter(new BufferedWriter(fout));
			generateFile(n ,seed, out);
			out.close();
		} catch (IOException e) {
			System.err.println(e);
			System.exit(1);
		}
		System.exit(0);
	}

}

File: examples/disk-IO-examples-C/generate_text.c

#include <stdio.h>
#include <stdlib.h>
#include <time.h>

struct record {
	int key;
	double value1;
	double value2;
	double value3;
};

const int MAX_KEY = 1000000;

void generate_file(int n, unsigned int seed, FILE *fout)
{
	int i;
	struct record *next;

	next = (struct record *) malloc(sizeof(struct record));

	srand(seed); /* set starting seed for random num. generator */
	for (i=0; i<n; i++) {
		next->key = rand() % MAX_KEY;
		next->value1 = rand()/((double)MAX_KEY);
		next->value2 = rand()/((double)MAX_KEY);
		next->value3 = rand()/((double)MAX_KEY);
		fprintf(fout, "%d %lf %lf %lf\n", next->key, 
				next->value1, next->value2, next->value3);
	}
}

int main(int argc, char **argv)
{
	int n;
	unsigned int seed = 0;
	FILE *fout;

	if (argc < 2) {
		fprintf(stderr, "Usage: %s <n> [<seed>]\n", argv[0]);
		exit(1);
	}

	n = atoi(argv[1]);
	if (argc == 3) {
		seed = atoi(argv[1]);
	} 


	fout = fopen("data.txt", "w");
	if (!fout) {
		perror("gentxt");
	}

	generate_file(n ,seed, fout);

	fclose(fout);
	exit(0);
}
